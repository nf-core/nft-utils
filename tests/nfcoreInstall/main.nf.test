nextflow_process {

    name "Test Process TEST_MODULE"
    script "./main.nf"
    process "TEST_MODULE"

    tag "test"

    test("test plugin") {

        setup {
            nfcoreSetup("${launchDir}/library/")
            nfcoreInstall("${launchDir}/library/", ["minimap2/index"])
        }

        then {
            assertAll(
                { assert path("${launchDir}/library/modules.json").exists()                                   },
                { assert path("${launchDir}/library/modules/nf-core/minimap2/index/main.nf").exists()                 },
                { assert path("${launchDir}/library/modules/nf-core/minimap2/index/environment.yml").exists()         },
                { assert path("${launchDir}/library/modules/nf-core/minimap2/index/meta.yml").exists()                },
                { assert path("${launchDir}/library/modules/nf-core/minimap2/index/tests/main.nf.test").exists()      },
                { assert path("${launchDir}/library/modules/nf-core/minimap2/index/tests/main.nf.test.snap").exists() }
            )
        }

    }

}
