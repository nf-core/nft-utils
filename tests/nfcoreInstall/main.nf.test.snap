{
    "test module map": {
        "content": [
            [
                "environment.yml",
                "main.nf",
                "meta.yml",
                "tests/main.nf.test",
                "tests/main.nf.test.snap"
            ],
            [
                "main.nf",
                "meta.yml"
            ],
            [
                "main.nf:md5,88aeb64bdd2bac0d334bb9abf51f410a",
                "meta.yml:md5,c6c37522e65307a75649c83fc1d718fc"
            ],
            [
                "environment.yml:md5,1ad01a097bd7a4bd7ac64ecad2c00c45",
                "main.nf:md5,6b63c5012f03241f9087e15b6152ccac",
                "meta.yml:md5,dc579195e211eaf80b35a36b3a58d36b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-26T09:43:44.521124"
    },
    "test module list": {
        "content": [
            [
                "environment.yml",
                "main.nf",
                "meta.yml",
                "tests/main.nf.test",
                "tests/main.nf.test.snap"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-26T09:43:24.564405"
    }
}